### PR TITLE
Rent model

### PR DESCRIPTION
**Pull request by micaelacole**
# Summary
A Rent model is created, with references to the User and Book models, and date-type fields for the start date and end date of the rental.

After creating the model with its migration, a Rent instance can be created within the Ruby console (irb). Since it references or _belongs_to_ certain book and user, it is necessary to create an instance of these two classes first. For example:

-  `User.create(first_name: "Test", last_name: "TestLastName", email: "test@wolox.com.ar", password: "123123123", password_confirmation: "123123123")`

- `Book.create(genre: "Fantasy", title: "The Chronicles of Narnia: The Lion, the Witch and the Wardrobe", author: "C. S. Lewis", cover_image: "narnia_cover.jpg", publisher: "Geoffrey Bles", year: "1950")`

- `Rent.create(user: User.first, book: Book.first, start_date: Date.new(2018, 8, 5), end_date: Date.new(2018, 8, 9))`

With those three tasks, a Rent instance can be created, returning: `#<Rent id: 1, user_id: 1, book_id: 1, start_date: "2018-08-05", end_date: "2018-08-09", created_at: "2018-10-30 20:11:04", updated_at: "2018-10-30 20:11:04">`

# Trello Card
https://trello.com/c/B98A8RY0

### CODE REVIEW

File: spec/factories/rents.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FactoryBot.define do
+  factory :rent do
+    user { nil }
+    book { nil }
+    start_date { 'MyString' }
```
Comment (nicozar95) [inline spec/factories/rents.rb]: Don't use constant in a factory

**Review by nicozar95 — CHANGES_REQUESTED** [review]

File: app/controllers/books_controller.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class BooksController < ApplicationController
+  def index
+    @books = Book.all
```
Comment (facucompe) [inline app/controllers/books_controller.rb]: There is no need to use a instance variable, you can do `render_paginated Book.all`

File: app/controllers/books_controller.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    render_paginated @books
+  end
+
+  def show
+    @book = Book.find(params[:id])
```
Comment (facucompe) [inline app/controllers/books_controller.rb]: Same here

File: spec/controllers/books_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  describe 'GET #show' do
+    context 'When fetching a book' do
+      let!(:book) { create(:book) }
+
+      before do
+        get :show, params: { id: book.id }
+      end
+
+      it 'responses with the book json' do
+        expect(response.body.to_json) =~ BookSerializer.new(
```
Comment (facucompe) [inline spec/controllers/books_controller_spec.rb]: Add this:
```
module Response
  module JSONParser
    def response_body
      ActiveSupport::JSON.decode(response.body) if response.present? && response.body.present?
    end
  end
end
```
So you can do `response_body`.

**Review by facucompe — CHANGES_REQUESTED** [review]

File: spec/factories/rents.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+FactoryBot.define do
+  factory :rent do
+    user
```
Comment (nicozar95) [inline spec/factories/rents.rb]: Why is the user and book empty on the Factory?

Comment (micaelacole) [reply spec/factories/rents.rb]: I looked up how to create a Factory of an object that references on other objects and found this solution. What I understood from it is that in this way we can call the User and Book factories in order to create the Rent factory. I think I tried doing FactoryBot.create(rent) and it worked, but I'm gonna try it one more time to be sure.

Comment (nicozar95) [reply spec/factories/rents.rb]: You can call directly to create a user in a FactoryBot, for example:
``` 
FactoryBot.define do
  factory :rent do
    user { create(:user) }
    ...
  end
end
```

File: app/controllers/books_controller.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+class BooksController < ApplicationController
```
Comment (nicozar95) [inline app/controllers/books_controller.rb:1]: What does this had to do with the card? 🤔

**Review by nicozar95 — CHANGES_REQUESTED** [review]
I think there it's a little of mixed features in this PR 😕

File: app/controllers/books_controller.rb (hunk 6 above)
Comment (micaelacole) [reply app/controllers/books_controller.rb:1]: I merged the other approved PRs into this one.

Comment (nicozar95) [reply app/controllers/books_controller.rb:1]: You should have rebase the changes

File: spec/models/rent_spec.rb [hunk 7]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
```
Comment (nicozar95) [inline spec/models/rent_spec.rb]: Remove this file

File: spec/controllers/rents_controller_spec.rb [hunk 8]
```diff
@@ -0,0 +1,4 @@
+require 'rails_helper'
```
Comment (nicozar95) [inline spec/controllers/rents_controller_spec.rb]: Remove this file... for now

**Review by nicozar95 — DISMISSED** [review]
Remove unecesary files and it's well to go

**Review by facucompe — APPROVED** [review]

**Pull request merged**